src/analysis/raster/qgsrastercalcnode.cpp

```diff
@@ -382,3 +382,21 @@ QgsRasterCalcNode *QgsRasterCalcNode::parseRasterCalcString( const QString &str,
   return localParseRasterCalcString( str, parserErrorMsg );
 }
 
+QStringList QgsRasterCalcNode::referencedLayerNames( const QString &str )
+{
+  QStringList referencedRasters;
+
+  QString errorString;
+  QgsRasterCalcNode *testNode = QgsRasterCalcNode::parseRasterCalcString( str, errorString );
+
+  const QList<const QgsRasterCalcNode *> rasterRefNodes =  testNode->findNodes( QgsRasterCalcNode::Type::tRasterRef );
+  for ( const QgsRasterCalcNode *r : rasterRefNodes )
+  {
+    QString layerRef( r->toString().remove( 0, 1 ) );
+    layerRef.chop( 1 );
```

Comment:
This code is to remove the double quotes, right? I think the syntax of expressions allows raster references also without quotes, so you should probably check first if the quotes are present before removing them...

Reply:
Ok, I added the check for the moment. Thank you :+1: